### ISSUE TITLE
Changing RDS from "standard" to "gp2" doesn't work

### ISSUE THREAD

**Issue by mdevs5531** [hashicorp/terraform#4281]
Seems like this is a bug, it says it applies the changes but nothing happens, and will try to apply the changes again every time apply is run.

To reproduce, create an RDS instance with no defined storage type:
```
resource "aws_db_instance" "db" {
    name = "testdb"
    identifier = "db"
    allocated_storage = 20
    engine = "postgres"
    engine_version = "9.4.5"
    instance_class = "db.t2.small"
    storage_type = "gp2"
    username = "me"
    password = "abcd1234"
    publicly_accessible = false
}
```

`terraform show` reports:
```
storage_type = standard
```

now add a definition for `storage_type = "gp2"`

`terraform plan` reports:
```
~ aws_db_instance.db
    storage_type: "standard" => "gp2"
```

then run `terraform apply`:
```
aws_db_instance.db: Refreshing state... (ID: db)
aws_db_instance.db: Modifying...
  storage_type: "standard" => "gp2"
aws_db_instance.db: Modifications complete

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

`terraform show` reports:
```
  storage_type = standard
```

and running `terraform plan` gives the same output as above.

**Comment by stack72** [comment]
@mdevs5531 I think in this case you will need to include the apply_immediately flag in your resource:

```
resource "aws_db_instance" "db" {
    name = "testdb"
    identifier = "db"
    allocated_storage = 20
    engine = "postgres"
    engine_version = "9.4.5"
    instance_class = "db.t2.small"
    storage_type = "gp2"
    username = "me"
    password = "abcd1234"
    publicly_accessible = false
    apply_immediately = true
}
```

Otherwise RDS is waiting for the next maintenance window for this to happen. 

**Please be careful here though, this may reboot your RDS Instance**

**Comment by jen20** [comment]
I wonder if we shouldn't introduce some kind of informative message as part of the application output warning when this manner or thing occurs? Any thoughts here @phinze?

**Comment by mdevs5531** [comment]
@stack72 Thanks, you're totally correct.

I was thinking it might help to have a note in the docs as well?  The behavior does make sense in a way, but is very unexpected given how there rest of terraform applies happen.  There is a DescribePendingMaintenanceActions API call, so in theory Terraform could inspect that and use the information somehow, but I can see that logic might get complicated.

**Comment by catsby** [comment]
Hey @mdevs5531 – I've added some additional docs in this PR: https://github.com/hashicorp/terraform/pull/4386

Using the `DescribePendingMaintenanceActions` API is definitely something we should consider using, though it's not yet clear how we could present that to the user. We'll look into it!

I'm going to mark this a "thinking" for now, let me know if you think there's more investigation warranted here.

Thanks!

**Comment by mdevs5531** [comment]
Sounds great to me, the doc updates would definitely have addressed my confusion.  Thanks for your help!

**Comment by stack72** [comment]
Closed via the docs additions in #4386
